### DISCUSSION TITLE
Add topography in gempy 3

### DISCUSSION THREAD

**Discussion by manthue** [gempy-project/gempy#903] [Q&A]
Dear all
I just started with gempy and noticed that adding the surface topography from a file (e.g. tif raster) is not yet implemented. My question: Is there a workaround this to have a topograpy anyway (I mean a defined topography from a given raster), e.g. by reading it by another python package and then passing it in as grid?

Thanks for your kind help, Manfred

**Comment by javoha** [comment]
Hi,
sorry for the inconvenience. There seems to be a dependency issue with subsurface. We will fix this in the near future. If you have a local clone of the repository there is a quick change you can make right now that worked for me (at least with a DEM). You need to have subsurface and rasterio installed.
In "gempy/API/grid_API.py" replace line 59 with
`struct: ss.StructuredData = ss.reader.read_structured_topography(`
Cheers
Jan

**Reply by manthue** [reply]
Hi javoha
Thanks for the quick reply, but I still have difficulties to make this work:
- So I cloned gempy from github to a local directory and installed the package with the modification you suggest. So far so good.
- However, when I try it in a jupyter notebook, upon importing gempy_viewer (which I would need), I get a "No module named 'gempy_viewer'" error.
- So I tried to install gempy from local with [opt] requirements (which worked with the non-dev version I used before).
- This however gives me errors due to conflicting dependencies. The only way I can install is with [dev], which apparently does not have gempy_viewer in it.

Do you have suggestions where I am going wrong? How can I have the version of gempy I need from a local install?
Thanks, Manfred

**Reply by javoha** [reply]
Hi again,
yes I see why this is a bit confusing. In version 3 gempy has multiple repositories (gempy, gempy_engine, gempy_viewer,...) to have these things separated. I am not the greates expert on proper git routines but here is what worked for me: 
1) Create an empty directory
2) Clone gempy (really the repo called gempy)
3) Create a new virtual environment (I used python 3.12 and the basic anaconda distribution)
4) pip install notebook, pip install gempy_viewer, pip install gempy_core (and whatever else you need e.g pyvista) but NOT gempy, as this would overwrite your local version
5) Start a notebook in the same directory
6) 
```
import os, sys
os.chdir(r"C:\Users\myname\Desktop\localgem") #In this case path to my local directory
sys.path.append("gempy")

import gempy as gp
import gempy_viewer as gpv
```

Then you have a local gempy version - ready to be manually fixed - and gempy_viewer and engine installed from pip.

Again, we will hopefully also have this fixed in the release version soonish.

**Comment by lucianomarquetto** [comment]
Hi all!

I would have asked the same question as @manthue (how to work around this topography issue), then saw this detailed response by @javoha. Just wanted to thank him and mention that this worked for me o/

A quick correction for beginners in gempy as I am: in step 4 (package installation) for me the "pip install gempy_core" did not work, I think the correct would be "pip install gempy_engine" (although if I remember correctly it was already installed). Another thing is that I also needed to install scipy ("pip install scipy") and subsurface ("pip install subsurface") for this to work. 

Anyway, thanks again for this, the topography makes a much better visualization of the model. Looking forward the stable solution!

Cheers!

Luciano.
